### PR TITLE
[v0.5] dex/candles: delta overflow bug

### PR DESCRIPTION
**Pull request by buck54321**
whoops

### CODE REVIEW

**Review by JoeGruffins — APPROVED** [review]

File: dex/candles/candles.go [hunk 1]
```diff
@@ -130,8 +130,8 @@ func (c *Cache) Delta(since time.Time) (changePct float64, vol uint64) {
 		if candle.StartStamp <= cutoff && candle.StartRate != 0 {
 			// Interpret the point linearly between the start and end stamps
 			cut := float64(cutoff-candle.StartStamp) / float64(candle.EndStamp-candle.StartStamp)
-			rateDelta := candle.EndRate - candle.StartRate
-			r := candle.StartRate + uint64(cut*float64(rateDelta))
+			rateDelta := float64(candle.EndRate) - float64(candle.StartRate)
```
Comment (chappjc) [inline dex/candles/candles.go:133]: Oh signed integers.

Trying to gauge impact.  There going to be candles in the epoch_reports with crazy spot rate changes, and there are going to be crazy candles from the `/candles` endpoint, but on restart will all that correct itself when `AddMarketSource` >  `LoadEpochStats` runs?

Comment (chappjc) [reply dex/candles/candles.go:133]: Can you also update `TestDelta` with a `Candle` where `EndRate` is less than `StartRate`?

Comment (buck54321) [reply dex/candles/candles.go:133]: > Oh unsigned integers.
> 
> Trying to gauge impact. There going to be candles in the epoch_reports with crazy spot rate changes, and there are going to be crazy candles from the `/candles` endpoint, but on restart will all that correct itself when `AddMarketSource` > `LoadEpochStats` runs?

That's right. Nothing is wrong in the DB, so everything should correct itself on restart.

**Review by chappjc — APPROVED** [review]

**Comment by chappjc** [comment]
OK, merged to `release-v0.5` and operators can simply deploy from that branch.  No client updates require AFAICT.

We'll have to put this on master too.

**Pull request merged**